cli-plugins/socket/socket.go

```diff
@@ -14,21 +14,23 @@ import (
 const EnvKey = "DOCKER_CLI_PLUGIN_SOCKET"
 
 // SetupConn sets up a Unix socket listener, establishes a goroutine to handle connections
-// and update the conn pointer, and returns the environment variable to pass to the plugin.
-func SetupConn(conn **net.UnixConn) (string, error) {
+// and update the conn pointer, and returns the environment variable to pass to the plugin
+// as well as the listener's close function
+func SetupConn(conn **net.UnixConn) (string, func(), error) {
 	listener, err := listen("docker_cli_" + uuid.Generate().String())
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	accept(listener, conn)
 
-	return EnvKey + "=" + listener.Addr().String(), nil
+	closeListener := func() {
+		listener.Close()
```

Comment:
Perhaps take the opportunity to make linters and IDEs "happy";

```suggestion
		_ = listener.Close()
```

Reply:
Oh! My linters don't complain about this, I wonder if it's Goland (maybe we can add something to `.golangci.yml` so that CI/people not using Goland would pick it up too). Btw, does it also complain about this in a `defer`?